core/src/components/button/button.tsx

```diff
@@ -295,6 +305,10 @@ export class Button implements ComponentInterface, AnchorInterface, ButtonInterf
     this.ionBlur.emit();
   };
 
+  private slotChanged = () => {
+    this.isCircle = this.hasIconOnly;
```

Comment:
~ We should explain that we are using a state variable to trigger a re-render of the Stencil component when the slotted content changes, so that `has-icon-only` class is correctly applied. 

I think moving that documentation here to enable the earlier documentation to be more focused on what `isCircle` is - a variable to track if the button should render as a circle shape or not.